src/com/Communication.cpp

```diff
@@ -286,5 +294,70 @@ void Communication::broadcast(std::vector<double> &v, int rankBroadcaster)
   broadcast(v.data(), size, rankBroadcaster);
 }
 
+void Communication::writeConnectionInfo(std::string const & acceptorName,
+                                        std::string const & requesterName,
+                                        int rank,
+                                        std::string addressDirectory,
+                                        std::string addressData)
+{
+  using namespace boost::filesystem;
+  int const firstLevelLen = 2;
+
+  boost::uuids::string_generator ns_gen;
+  auto ns = ns_gen("af7ce8f2-a9ee-46cb-38ee-71c318aa3580"); // md5 hash of precice.org as namespace
+  
+  boost::uuids::name_generator gen{ns};
+  std::string s = acceptorName + requesterName + std::to_string(rank);
```

Comment:
As @uekerman stated on the mailing list: To make this future proof, we need the name of the mesh in here too.

Reply:
And as I replied on the mailing list, this needs some more invasive changes. Because of that, I do first things first (replace the current system) and fix potential, upcoming problems after that.

Reply:
@floli This is just for the record as this is an actual PR.